agent/consul/discoverychain/gateway_tcproute.go

```diff
@@ -0,0 +1,17 @@
+package discoverychain
+
+import "github.com/hashicorp/consul/agent/structs"
+
+func synthesizeTCPRouteDiscoveryChain(route structs.TCPRouteConfigEntry) []structs.IngressService {
+	services := []structs.IngressService{}
```

Comment:
#nit we know what size this needs to be
```suggestion
	services := make([]structs.IngressService, 0, len(route.Services))
```